components/service/glean/src/test/java/mozilla/components/service/glean/storages/StringListsStorageEngineTest.kt

```diff
@@ -0,0 +1,181 @@
+                    clearStore = false)
+            assertEquals(1, snapshot!!.size)
+            assertEquals("First", snapshot.get("telemetry.string_list_metric")?.get(0))
+            assertEquals("Second", snapshot.get("telemetry.string_list_metric")?.get(1))
+        }
+    }
+
+    @Test
+    fun `add() properly adds the value in all stores`() {
+        val storeNames = listOf("store1", "store2")
+
+        val metric = StringListMetricType(
+                disabled = false,
+                category = "telemetry",
+                lifetime = Lifetime.Ping,
+                name = "string_list_metric",
+                sendInPings = storeNames
+        )
+
+        StringListsStorageEngine.add(
+                metricData = metric,
+                value = listOf("First")
+        )
+
+        // Check that the data was correctly added in each store.
+        for (storeName in storeNames) {
+            val snapshot = StringListsStorageEngine.getSnapshot(
+                    storeName = storeName,
+                    clearStore = false)
+            assertEquals("First", snapshot!!["telemetry.string_list_metric"]?.get(0))
+        }
+    }
+
+    @Test
+    fun `add() won't allow adding beyond the max list length`() {
+        val metric = StringListMetricType(
+                disabled = false,
+                category = "telemetry",
+                lifetime = Lifetime.Ping,
+                name = "string_list_metric",
+                sendInPings = listOf("store1")
+        )
+
+        for (i in 1..52) {
+            StringListsStorageEngine.add(
+                    metricData = metric,
+                    value = listOf("value$i")
+            )
+        }
+
+        // Check that list was truncated.
+        val snapshot = StringListsStorageEngine.getSnapshot(
+                storeName = "store1",
+                clearStore = false)
+        assertEquals(1, snapshot!!.size)
+        assertEquals(true, snapshot.containsKey("telemetry.string_list_metric"))
+        assertEquals(
+                StringListsStorageEngineImplementation.MAX_LIST_LENGTH_VALUE,
+                snapshot["telemetry.string_list_metric"]?.count()
+        )
+    }
+
+    @Test
+    fun `set() won't allow adding a list longer than the max list length`() {
+        val metric = StringListMetricType(
+                disabled = false,
+                category = "telemetry",
+                lifetime = Lifetime.Ping,
+                name = "string_list_metric",
+                sendInPings = listOf("store1")
+        )
+
+        val stringList: MutableList<String> = mutableListOf()
+        for (i in 1..StringListsStorageEngineImplementation.MAX_LIST_LENGTH_VALUE + 1) {
+            stringList.add("value$i")
+        }
+
+        StringListsStorageEngine.set(metricData = metric, value = stringList)
+
+        // Check that list was truncated.
+        val snapshot = StringListsStorageEngine.getSnapshot(
+                storeName = "store1",
```

Comment:
nit: this looks like it should make klint fail. Does it pass?

Reply:
It does, I'm not sure why my editor indents this way but ktlint/detekt seem to allow it.

Reply:
Oh, fine then :)

Reply:
Since this wasn't a choice and was simply an editor formatting default, I went through and made all of the indentations a single indent to be more uniform.

Reply:
Cool! In general, you should write code in a style that matches the style of other files within the same repo. Matching the style of stuff  in `glean` would do :)